alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ChannelDistributionUIConfig.java

```diff
@@ -50,48 +46,61 @@
     }
 
     @Override
     public List<ConfigField> createFields() {
+        final ConfigField channelName = SelectConfigField.createRequired(KEY_CHANNEL_NAME, LABEL_CHANNEL_NAME, DESCRIPTION_CHANNEL_NAME, getChannelLabelValues());
         final ConfigField name = TextInputConfigField.createRequired(KEY_NAME, LABEL_NAME, DESCRIPTION_NAME);
         final ConfigField frequency = SelectConfigField.createRequired(KEY_FREQUENCY, LABEL_FREQUENCY, DESCRIPTION_FREQUENCY, Arrays.stream(FrequencyType.values())
                                                                                                                                   .map(frequencyType -> new LabelValueSelectOption(frequencyType.getDisplayName(), frequencyType.name()))
                                                                                                                                   .collect(Collectors.toList()));
-        final ConfigField channelName = SelectConfigField.createRequired(KEY_CHANNEL_NAME, LABEL_CHANNEL_NAME, DESCRIPTION_CHANNEL_NAME, getDescriptorNames(DescriptorType.CHANNEL));
-        final ConfigField providerName = SelectConfigField.createRequired(KEY_PROVIDER_NAME, LABEL_PROVIDER_NAME, DESCRIPTION_PROVIDER_NAME, getDescriptorNames(DescriptorType.PROVIDER));
 
-        final List<ConfigField> configFields = List.of(name, channelName, frequency, providerName);
+        final ConfigField providerName = SelectConfigField.createRequired(KEY_PROVIDER_NAME, LABEL_PROVIDER_NAME, DESCRIPTION_PROVIDER_NAME, getProviderLabelValues());
+
+        final List<ConfigField> configFields = List.of(channelName, name, frequency, providerName);
         final List<ConfigField> channelDistributionFields = createChannelDistributionFields();
         return Stream.concat(configFields.stream(), channelDistributionFields.stream()).collect(Collectors.toList());
     }
 
     public abstract List<ConfigField> createChannelDistributionFields();
 
-    private Collection<LabelValueSelectOption> getDescriptorNames(final DescriptorType descriptorType) {
-        try {
-            return descriptorAccessor.getRegisteredDescriptorsByType(descriptorType).stream()
-                       .map(RegisteredDescriptorModel::getName)
-                       .map(LabelValueSelectOption::new)
-                       .collect(Collectors.toSet());
-        } catch (final AlertDatabaseConstraintException e) {
-            logger.error("There was an error when retrieving data from the DB when building fields.");
-        }
-        return Set.of();
+    public String getChannelName() {
+        return channelName;
+    }
+
+    private Collection<LabelValueSelectOption> getProviderLabelValues() {
+        return descriptorMap.getDescriptorByType(DescriptorType.PROVIDER).stream()
+                   .map(descriptor -> descriptor.getMetaData(ConfigContextEnum.DISTRIBUTION))
+                   .filter(Optional::isPresent)
+                   .map(Optional::get)
```

Comment:
The lines 
```java
.filter(Optional::isPresent)
.map(Optional::get)
```
should become
```java
.flatMap(Optional::stream)
```

Reply:
Oh wow, I didn't think of that as an alternative.